CHANGELOG.md

```diff
@@ -2,6 +2,55 @@
 
 ## Unreleased
 
+### API
+
+#### API Extensions
+
+* DEPRECATION: the `opentelemetry-extension-annotations` module containing `@WithSpan`
+  and `@SpanAttribute` annotations has been deprecated for removal in next major version. A copy of
+  the code will instead be maintained
+  in [opentelemetry-java-instrumentation/instrumentation-annotations](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation-annotations)
+  and published under
+  coordinates `io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations:{version}`.
+
+### SDK
+
+#### Traces
+
+* Add default implementation for `SpanData#getInstrumentationScopeInfo()`
+  and `ReadableSpan#getInstrumentationScopeInfo()`. This fixes a previous mistake where those
+  interfaces were extended without default implementation, a breaking change for source
+  compatibility. Stricter checks have been added to ensure this mistake is not repeated.
+
+#### Logs
+
+* BREAKING: delete `LogDataBuilder`. A similar implementation of `LogData` called `TestLogData` has
+  been added to `opentelemetry-sdk-logs-testing`.
+* BREAKING: rename `LogProcessor#emit(LogData)` to `LogProcessor#onEmit(ReadWriteLogRecord)`. The
+  argument change from `LogData` to `ReadWriteLogRecord` allows implementations to mutate logs. To
+  obtain `LogData`, call `ReadWriteLogRecord#toLogData()`.
+* Optimize `SdkLogEmitterProvider` to return noop `LogEmitter` when no `LogProcessor`s are
+  registered.
+
+#### Exporter
```

Comment:
Is it necessary to include in the changelog these changes as they are noticeable by users:
* https://github.com/open-telemetry/opentelemetry-java/commit/d84a111273801a3b1be13dc7312379a2fc7266f6
* https://github.com/open-telemetry/opentelemetry-java/commit/4bfe17fb76c291f774de30e53cd3698e89e50adc

Reply:
I omitted those because I think a very small percentage of users will notice them.

Reply:
I think I'd err on the side of over-communicating in this case, but I don't feel super strongly about it.